File: middleware/auth_middleware.py
from fastapi import Request, HTTPException, status
from config import SECRET_KEY
import jwt

# This middleware is created but never actually used in the application
# Missing authentication middleware on routes

async def verify_token(request: Request):
    """
    Verify JWT token from request headers.
    This function is defined but never called.
    """
    # Hardcoded secret key usage
    token = request.headers.get("Authorization")
    
    if not token:
        raise HTTPException(
            status_code=status.HTTP_401_UNAUTHORIZED,
            detail="Missing authentication token"
        )
    
    try:
        # Hardcoded secret key
        payload = jwt.decode(token, SECRET_KEY, algorithms=["HS256"])
        return payload
    except jwt.InvalidTokenError:
        raise HTTPException(
            status_code=status.HTTP_401_UNAUTHORIZED,
            detail="Invalid authentication token"
        )

File: middleware/authMiddleware.js
const jwt = require('jsonwebtoken');

// Hardcoded JWT secret - should be in environment variables
const JWT_SECRET = 'my-secret-key-12345';

// This middleware is created but never actually used in routes
exports.verifyToken = (req, res, next) => {
  // Missing error handling
  const token = req.headers.authorization?.split(' ')[1];

  if (!token) {
    // Improper HTTP status code - should be 401 Unauthorized
    return res.status(403).json({ message: 'No token provided' });
  }

  // Missing try/catch block
  const decoded = jwt.verify(token, JWT_SECRET);
  req.userId = decoded.userId;
  next();
};

